cocos/audio/apple/AudioEngine-inl.mm

```diff
@@ -200,10 +200,14 @@ -(void) dealloc
         audioCache = &_audioCaches[filePath];
         audioCache->_fileFullPath = FileUtils::getInstance()->fullPathForFilename(filePath);
         
-        AudioEngine::addTask(std::bind(&AudioCache::readDataTask, audioCache));
+        AudioEngine::addTask(std::bind(&AudioCache::readDataTask, audioCache), callback);
     }
     else {
         audioCache = &it->second;
+        if(callback && audioCache->_alBufferReady)
+        {
+            callback(true);
```

Comment:
`callback(false)` is called in `AudioEngineThreadPool::threadFunc` when `task()` returns false.